app/experimenter/experiments/forms.py

```diff
@@ -42,6 +44,30 @@ def clean(self, value):
         return cleaned_value
 
 
+class DSIssueURLField(forms.URLField):
+
+    def clean(self, value):
+        cleaned_value = super().clean(value)
+
+        if cleaned_value:
+            err_str = "Please Provide a Valid URL ex: {ds_url}DS-123 or {ds_url}DO-123"
+
+            DS_root = urljoin(settings.DS_ISSUE_HOST, "DS-")
+            DO_root = urljoin(settings.DS_ISSUE_HOST, "DO-")
+
+            if (
+                DS_root not in cleaned_value and DO_root not in cleaned_value
+            ) or self.get_ds_issue_id(cleaned_value) is None:
+
+                raise forms.ValidationError(err_str.format(ds_url=settings.DS_ISSUE_HOST))
+        return cleaned_value
+
+    def get_ds_issue_id(self, bug_url):
+        ds = re.match(re.escape(settings.DS_ISSUE_HOST) + r"(DS|DO)-(\w+.*)", bug_url)
+
+        return ds.group(1)
```

Comment:
Wait what am I thinking, couldn't you do this whole thing with just the regex match looking for the ID and if match is None then we know it's invalid?

Reply:
being the regex newbie, what do you mean by regex match looking for the ID, like if the url ends in numbers?

Reply:
I mean like if `ds is None`, then the regex failed to find the ID at the end of the URL because it didn't match the regex, so then we know whatever URL was provided isn't a 'valid' jira URL, which accomplishes the same thing as the two `not in` checks above it, so you should be able to remove them and just do the regex check?